engine/src/main/java/io/zeebe/engine/processing/bpmn/gateway/ExclusiveGatewayProcessor.java

```diff
@@ -138,22 +128,14 @@ public void onEventOccurred(
         "Expected to handle occurred event on exclusive gateway, but events should not occur on exclusive gateway.");
   }
 
-  private ExecutableSequenceFlow getOutgoingSequenceFlow(
-      final ExecutableExclusiveGateway element,
-      final BpmnElementContext context,
-      final IndexedRecord record) {
-
-    final var sequenceFlowId = record.getValue().getElementIdBuffer();
-
-    return element.getOutgoing().stream()
-        .filter(sequenceFlow -> sequenceFlow.getId().equals(sequenceFlowId))
-        .findFirst()
-        .orElseThrow(
-            () ->
-                new IllegalStateException(
-                    String.format(
-                        "Expected sequence flow with id '%s' but not found. [context: %s]",
-                        bufferAsString(sequenceFlowId), context)));
+  private BpmnElementContext transitionToCompleted(final BpmnElementContext activating) {
+    Preconditions.checkArgument(
+        activating.getIntent() == WorkflowInstanceIntent.ELEMENT_ACTIVATING,
+        TRANSITION_TO_COMPLETED_PRECONDITION_ERROR,
+        activating.getIntent());
```

Comment:
This is the first usage of `Preconditions` outside of Atomix. In other places, we use our own `io.zeebe.util.EnsureUtil` or `Objects`. 

We should discuss it with the team and document the decision in the style guide.

Reply:
We'll leave this for now. Topic is moved to next team meeting